src/Microsoft.ML.Core/Data/ReadOnlyMemoryUtils.cs

```diff
@@ -254,5 +254,18 @@ public static StringBuilder AppendSpan(this StringBuilder sb, ReadOnlySpan<char>
 
             return sb;
         }
+
+        public sealed class ReadonlyMemoryCharComparer : IEqualityComparer<ReadOnlyMemory<char>>
+        {
+            public bool Equals(ReadOnlyMemory<char> x, ReadOnlyMemory<char> y)
+            {
+                return x.Span.SequenceEqual(y.Span);
+            }
+
+            public int GetHashCode(ReadOnlyMemory<char> obj)
+            {
+                return obj.ToString().GetHashCode();
```

Comment:
This is not a good idea, since it will allocate a new string every time GetHashCode is called.

I think we should use our Murmur hash algorithm to get a hash code instead.